modules/core/src/main/java/org/apache/ignite/cache/query/TextQuery.java

```diff
@@ -79,6 +82,19 @@ public TextQuery(String type, String txt) {
         setText(txt);
     }
 
+    /**
+     * Constructs query for the given search string.
+     *
+     * @param type Type.
+     * @param txt Search string.
+     * @param limit Limits response records count. If 0 or less, the limit considered to be Integer.MAX_VALUE, that is virtually no limit.
+     */
+    public TextQuery(String type, String txt, int limit) {
```

Comment:
'Limit' parameter is important part of Queries in Lucene text search. This is because responses usually contain many records and only certain relevant top of them interests. That's why limits are used in most cases when working with Lucene. This differs TextQuery from SQL ones.